sound/soc/sof/compress.c

```diff
@@ -227,12 +229,21 @@ static int sof_compr_set_params(struct snd_soc_component *component,
 
 	memcpy((u8 *)pcm->params.ext_data, &params->codec, ext_data_size);
 
-	ret = sof_ipc_tx_message_no_reply(sdev->ipc, pcm, sizeof(*pcm) + ext_data_size);
+	ret = sof_ipc_tx_message(sdev->ipc, pcm, sizeof(*pcm) + ext_data_size,
+				 &ipc_params_reply, sizeof(ipc_params_reply));
 	if (ret < 0) {
 		dev_err(component->dev, "error ipc failed\n");
 		goto out;
 	}
 
+	ret = snd_sof_set_stream_data_offset(sdev, &spcm->stream[cstream->direction],
+					     ipc_params_reply.posn_offset);
+	if (ret < 0) {
+		dev_err(component->dev, "Invalid stream data offset for Compr %d\n",
+			spcm->pcm.pcm_id);
+		goto out;
+	}
+
```

Comment:
Humm, not sure I get the need for this @dbaluta @iuliana-prodan or what posn_offset represents for a compressed stream.

Is this to track the DMA position? If yes, what happens for e.g. HDaudio where the position is not provided with any IPC but directly read from hardware registers (DPIB).

Reply:
Thanks @dbaluta I still have zero understanding of how this is used ...